games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,94 @@
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media \
+players. It enables you to run classic games on a wide range of computers and \
+consoles through its slick graphical interface. It has advanced features like \
+shaders, netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com/"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_FILENAME="retroarch-$portVersion.tar.gz"
+SOURCE_DIR="RetroArch-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg auto-merge
+	"
+
+USER_SETTINGS_FILES="
+	$relativeSettingsDir/retroarch/retroarch.cfg template \
+		$sysconfDir/retroarch.cfg
+	"
+
+PROVIDES="
+	retroarch$secondaryArchSuffix = $portVersion
+	app:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch_cg2glsl$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libavcodec$secondaryArchSuffix
+	lib:libavformat$secondaryArchSuffix
+	lib:libavdevice$secondaryArchSuffix
+	lib:libswresample$secondaryArchSuffix
+	lib:libavresample$secondaryArchSuffix
+	lib:libavutil$secondaryArchSuffix
+	lib:libswscale$secondaryArchSuffix
+	retroarch_assets
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:which
+	"
+
+PATCH()
+{
+	libretroDir="`finddir B_SYSTEM_DIRECTORY`/$relativeAddOnsDir/libretro"
+	assetsDir="`finddir B_SYSTEM_DATA_DIRECTORY`/retroarch/assets"
+	cacheDir="`finddir B_SYSTEM_CACHE_DIRECTORY`/retroarch"
+	contentDir="`finddir B_SYSTEM_VAR_DIRECTORY`/retroarch"
+	coreAssetsDir="`finddir B_SYSTEM_VAR_DIRECTORY`/retroarch/assets"
+	sed -i \
+		-e "s|^# \(libretro_directory =\)|\1 \"$libretroDir/\"|;" \
+		-e "s|^# \(libretro_info_path =\)|\1 \"$libretroDir/\"|;" \
+		-e "s|^# \(assets_directory =\)|\1 \"$assetsDir/\"|;" \
+		-e "s|^# \(cache_directory =\)|\1 \"$cacheDir/\"|;" \
+		-e "s|^# \(content_directory =\)|\1 \"$contentDir/\"|;" \
+		-e "s|^# \(core_assets_directory =\)|\1 \"$coreAssetsDir/\"|;" \
+		retroarch.cfg
+}
+
+BUILD()
+{
+	./configure --prefix="$prefix" --global-config-dir="$sysconfDir" \
+		--bindir="$appsDir" --datarootdir="$dataRootDir" \
+		--docdir="$docDir" --mandir="$manDir" \
+		--with-assets_dir="$dataDir"
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	addAppDeskbarSymlink "$appsDir"/retroarch RetroArch
+	install -m 0755 -d "$dataDir"/retroarch
```

Comment:
We **_might_** need this:
```diff
-	install -m 0755 -d "$dataDir"/retroarch 
+	install -m 0755 -d "$dataDir"/retroarch "$sharedStateDir"/retroarch
```
That will make the package have a `var/retroarch/` directory. But I think this won't be enough.
We will probably need to add something like `create-retroarch-directories.sh` in `additional-files`.
(That's what we do for `www-servers/lighttpd` with a `create-lighttpd-directories.sh`.)
But there might be a better method. (I think we should investigate before the merge.)

Reply:
Creating the directory there works, just remember to specify it in `GLOBAL_WRITABLE_FILES`